### ISSUE TITLE
--with-mzscheme does not compile vim 8 with mzscheme

### ISSUE THREAD

**Issue by todd-dsm** [Homebrew/homebrew-core#8716]
Hey guys, just noticed this today. I don't need mzscheme but someone might, someday. Brew seems fine; this seems to be the Vim 8 formula.

I'm working with a test system on a fresh install. The requested steps:
```bash
$ brew update
Already up-to-date.

$ brew doctor 
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in the App Store.

Warning: Putting non-prefixed coreutils in your path can cause gmp builds to fail.

Warning: Putting non-prefixed findutils in your path can cause python builds to fail.
```

### Bug reports:
The config
```bash
$ brew config 
HOMEBREW_VERSION: >1.1.0 (no git repository)
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9cce3414fbd39d146d97911dbb3bf07fffaa1080
Last commit: 27 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c0e2a9033c42e0911f87a24c8960dca9d80890f4
Core tap last commit: 48 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: single-core 64-bit ivybridge
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.11.0 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: N/A
macOS: 10.12-x86_64
Xcode: N/A
CLT: 8.0.0.0.1.1472435881
X11: N/A
```

The Steps and output:
```bash
brew install vim --override-system-vi --disable-nls --with-lua --with-mzscheme --with-tcl
Warning: vim: --disable-nls was deprecated; using --without-nls instead!
Warning: vim: --override-system-vi was deprecated; using --with-override-system-vi instead!
==> Using the sandbox
==> Downloading https://github.com/vim/vim/archive/v8.0.0165.tar.gz
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/vim-8.0.0165.tar.gz
==> ./configure --prefix=/usr/local --mandir=/usr/local/Cellar/vim/8.0.0165/share/man --enable-multibyte --with-tlib=ncurses --ena
==> make
==> make install prefix=/usr/local/Cellar/vim/8.0.0165 STRIP=/usr/bin/true
🍺  /usr/local/Cellar/vim/8.0.0165: 1,400 files, 22.2M, built in 1 minute 54 seconds

$ vim --version | egrep 'mzschem'
+byte_offset     -gettext         -mzscheme        +textobjects
```

**Comment by ilovezfs** [comment]
This should be fixed by #10130 if you don't mind testing it.

**Comment by ilovezfs** [comment]
Scratch that. Since we don't actually directly provide an mzscheme formula, fixing this is really beyond Homebrew's scope, and I'm going to remove the option. Thank you for reporting this.
